internal/guest/amdsev/report.go

```diff
@@ -0,0 +1,275 @@
+// MSG_REPORT_REQ Table 20.
+type reportRequest struct {
+	ReportData [64]byte
+	VMPL       uint32
+	_          [28]byte
+}
+
+// report is an internal representation of SEV-SNP report
+// https://www.amd.com/system/files/TechDocs/56860.pdf
+// ATTESTATION_REPORT Table 21.
+type report struct {
+	Version          uint32
+	GuestSVN         uint32
+	Policy           uint64
+	FamilyID         [16]byte
+	ImageID          [16]byte
+	VMPL             uint32
+	SignatureAlgo    uint32
+	PlatformVersion  uint64
+	PlatformInfo     uint64
+	AuthorKeyEn      uint32
+	Reserved1        uint32
+	ReportData       [64]byte
+	Measurement      [48]byte
+	HostData         [32]byte
+	IDKeyDigest      [48]byte
+	AuthorKeyDigest  [48]byte
+	ReportID         [32]byte
+	ReportIDMA       [32]byte
+	ReportTCB        uint64
+	Reserved2        [24]byte
+	ChipID           [64]byte
+	CommittedSVN     [8]byte
+	CommittedVersion [8]byte
+	LaunchSVN        [8]byte
+	Reserved3        [168]byte
+	Signature        [512]byte
+}
+
+func (sr *report) report() Report {
+	return Report{
+		Version:          sr.Version,
+		GuestSVN:         sr.GuestSVN,
+		Policy:           sr.Policy,
+		FamilyID:         hex.EncodeToString(mirrorBytes(sr.FamilyID[:])[:]),
+		ImageID:          hex.EncodeToString(mirrorBytes(sr.ImageID[:])[:]),
+		VMPL:             sr.VMPL,
+		SignatureAlgo:    sr.SignatureAlgo,
+		PlatformVersion:  sr.PlatformVersion,
+		PlatformInfo:     sr.PlatformInfo,
+		AuthorKeyEn:      sr.AuthorKeyEn,
+		ReportData:       hex.EncodeToString(sr.ReportData[:]),
+		Measurement:      hex.EncodeToString(sr.Measurement[:]),
+		HostData:         hex.EncodeToString(sr.HostData[:]),
+		IDKeyDigest:      hex.EncodeToString(sr.IDKeyDigest[:]),
+		AuthorKeyDigest:  hex.EncodeToString(sr.AuthorKeyDigest[:]),
+		ReportID:         hex.EncodeToString(sr.ReportID[:]),
+		ReportIDMA:       hex.EncodeToString(sr.ReportIDMA[:]),
+		ReportTCB:        sr.ReportTCB,
+		ChipID:           hex.EncodeToString(sr.ChipID[:]),
+		CommittedSVN:     hex.EncodeToString(sr.CommittedSVN[:]),
+		CommittedVersion: hex.EncodeToString(sr.CommittedVersion[:]),
+		LaunchSVN:        hex.EncodeToString(sr.LaunchSVN[:]),
+		Signature:        hex.EncodeToString(sr.Signature[:]),
+	}
+}
+
+// reportResponse is the attestation response struct
+// https://www.amd.com/system/files/TechDocs/56860.pdf
+// MSG_REPORT_RSP Table 23.
+// NOTE: reportResponse.Report is a byte slice, to have the original
+//     response in bytes. The conversion to internal struct happens inside
+//     convertRawReport.
+// NOTE: the additional 64 bytes are reserved, without them, the ioctl fails.
+type reportResponse struct {
+	Status     uint32
+	ReportSize uint32
+	reserved1  [24]byte
+	Report     [1184]byte
+	reserved2  [64]byte // padding to the size of SEV_SNP_REPORT_RSP_BUF_SZ (i.e., 1280 bytes)
+}
+
+// FetchRawSNPReport returns attestation report bytes.
+func FetchRawSNPReport(reportData string) ([]byte, error) {
```

Comment:
My earlier question now extends to: do we actually ever pass something for `reportData` here? If not should we just leave it off?

Reply:
@svolos do you think this will be useful for the SKR code? If that's the case, I'll just export this to `/pkg/amdsev` and add the `reportData` field back.